crates/fuel-core/src/database.rs

```diff
@@ -411,6 +415,65 @@ impl fuel_core_storage::vm_storage::VmStorageRequirements for Database {
     }
 }
 
+impl AtomicView for Database {
+    type View = Database;
+
+    type Height = BlockHeight;
+
+    fn latest_height(&self) -> BlockHeight {
+        // TODO: The database should track the latest height inside of the database object
+        //  instead of fetching it from the `FuelBlocks` table. As a temporary solution,
+        //  fetch it from the table for now.
+        self.latest_height().unwrap_or_default()
+    }
+
+    fn view_at(&self, _: &BlockHeight) -> StorageResult<Self::View> {
+        // TODO: Unimplemented until of the https://github.com/FuelLabs/fuel-core/issues/451
+        Ok(self.latest_view())
+    }
+
+    fn latest_view(&self) -> Self::View {
+        // TODO: https://github.com/FuelLabs/fuel-core/issues/1581
+        self.clone()
+    }
+}
+
+pub struct RelayerReadDatabase(Database);
+
+impl RelayerReadDatabase {
+    pub fn new(database: Database) -> Self {
+        Self(database)
+    }
+}
+
+impl AtomicView for RelayerReadDatabase {
+    type View = Database;
+    type Height = DaBlockHeight;
+
+    fn latest_height(&self) -> Self::Height {
+        #[cfg(feature = "relayer")]
+        {
+            use fuel_core_relayer::ports::RelayerDb;
+            // TODO: The database should track the latest da height inside of the database object
```

Comment:
Is there a follow-up issue fore this?

Reply:
Yep, https://github.com/FuelLabs/fuel-core/issues/1589